### PR TITLE
Fix support to mongodb v5

### PR DESCRIPTION
**Pull request by mptr**
### Description of change

Support for `mongodb@^5.2.0`
As Requested in [this Comment](https://github.com/typeorm/typeorm/pull/9778#issuecomment-1498631152) on #9778 this PR includes a copy of the mongodb typings of version 5.2.0. (including typings for bson 5.2.0).
Occourences of `ObjectID` have been renamed to `ObjectId` to match the type name in [`bson`](https://github.com/mongodb/js-bson/blob/0d49a636052b540a8fc7bd9210db33a8a52f951e/src/objectid.ts#L31).

This contribution is entirely based on #9778. (Therefore closes #9778)
Thanks to @MatheusMeloAntiquera and @tgrassl for the good work!

### Pull-Request Checklist

- [X] Code is up-to-date with the `master` branch
- [X] `npm run format` to apply prettier formatting
- [X] `npm run test` passes with this change
  &rarr; Tested against sqlite, postgres, mongodb
- [X] This pull request links relevant issues as `Fixes #0000`
- [X] There are new or updated unit tests validating the change
- [X] Documentation has been updated to reflect this change
- [X] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/github-issues/7907/issue-7907.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { expect } from "chai"
+import { Connection } from "../../../src"
+import {
+    closeTestingConnections,
+    createTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { Post } from "./entity/Post"
+
+describe("github issues > #7907 add support for mongodb driver v4", () => {
```
Comment (pleerock) [inline test/github-issues/7907/issue-7907.ts:11]: v5

**Comment by pleerock** [comment]
Thank you for contribution! Can you please check why tests are failing?

**Comment by mptr** [comment]
@pleerock sure, will check today

**Comment by mptr** [comment]
@pleerock as far as i can tell, the tests fail because
- #9926 is not merged yet
- `ORA-12514` which i suspect to be an error not caused by changes in this pr. [see here](https://i.redd.it/w9rkv4ampbta1.jpg)

**Comment by pleerock** [comment]
Let's keep it. Thank you guys @mptr @MatheusMeloAntiquera @tgrassl for your efforts!

**Pull request merged**